tools/distrib/gen_compilation_database.py

```diff
@@ -0,0 +1,130 @@
+# https://github.com/envoyproxy/envoy/blob/master/tools/gen_compilation_database.py
+
+import argparse
+import glob
+import json
+import logging
+import os
+import re
+import shlex
+import subprocess
+from pathlib import Path
+
+RE_INCLUDE_SYSTEM = re.compile("\s*-I\s+/usr/[^ ]+")
+
+
+# This method is equivalent to https://github.com/grailbio/bazel-compilation-database/blob/master/generate.sh
+def generateCompilationDatabase(args):
+    # We need to download all remote outputs for generated source code. This option lives here to override those
+    # specified in bazelrc.
+    bazel_options = shlex.split(os.environ.get("BAZEL_BUILD_OPTIONS", "")) + [
+        "--config=compdb",
+        "--remote_download_outputs=all",
+    ]
+
+    subprocess.check_call(["bazel", "build"] + bazel_options + [
+        "--aspects=@bazel_compdb//:aspects.bzl%compilation_database_aspect",
+        "--output_groups=compdb_files,header_files"
+    ] + args.bazel_targets)
+
+    execroot = subprocess.check_output(["bazel", "info", "execution_root"] +
+                                       bazel_options).decode().strip()
+
+    compdb = []
+    for compdb_file in Path(execroot).glob("**/*.compile_commands.json"):
+        compdb.extend(
+            json.loads(
+                "[" +
+                compdb_file.read_text().replace("__EXEC_ROOT__", execroot) +
+                "]"))
+    return compdb
+
+
+def isHeader(filename):
+    for ext in (".h", ".hh", ".hpp", ".hxx"):
+        if filename.endswith(ext):
+            return True
+    return False
+
+
+def isCompileTarget(target, args):
+    filename = target["file"]
+    if not args.include_headers and isHeader(filename):
+        return False
+    if not args.include_genfiles:
+        if filename.startswith("bazel-out/"):
+            return False
+    if not args.include_external:
+        if filename.startswith("external/"):
+            return False
+    return True
+
+
+def modifyCompileCommand(target, args):
+    cc, options = target["command"].split(" ", 1)
+
+    # Workaround for bazel added C++11 options, those doesn't affect build itself but
+    # clang-tidy will misinterpret them.
+    options = options.replace("-std=c++0x ", "")
+    options = options.replace("-std=c++11 ", "")
+
+    if args.vscode:
+        # Visual Studio Code doesn't seem to like "-iquote". Replace it with
+        # old-style "-I".
+        options = options.replace("-iquote ", "-I ")
+
+    if args.ignore_system_headers:
+        if target["file"] == "test/cpp/end2end/xds_end2end_test.cc":
+            # Remove all include options for /usr/* directories
+            options = RE_INCLUDE_SYSTEM.sub("", options)
+
+    if isHeader(target["file"]):
+        options += " -Wno-pragma-once-outside-header -Wno-unused-const-variable"
+        options += " -Wno-unused-function"
+        if not target["file"].startswith("external/"):
+            # *.h file is treated as C header by default while our headers files are all C++17.
```

Comment:
Why is this needed?  And why C++17?  Aren't we still requiring C++11 support?

Reply:
Yeah C++11 would be the right one